app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -1443,13 +1461,30 @@ DeviceDelegateOculusVR::EndFrame(const bool aDiscard) {
   projection.HeadPose = m.predictedTracking.HeadPose;
   projection.Header.SrcBlend = VRAPI_FRAME_LAYER_BLEND_ONE;
   projection.Header.DstBlend = VRAPI_FRAME_LAYER_BLEND_ONE_MINUS_SRC_ALPHA;
-  for (int i = 0; i < VRAPI_FRAME_LAYER_EYE_MAX; ++i) {
-    const auto &eyeSwapChain = m.eyeSwapChains[i];
-    const int swapChainIndex = m.frameIndex % eyeSwapChain->swapChainLength;
-    // Set up OVR layer textures
-    projection.Textures[i].ColorSwapChain = eyeSwapChain->ovrSwapChain;
-    projection.Textures[i].SwapChainIndex = swapChainIndex;
-    projection.Textures[i].TexCoordsFromTanAngles = ovrMatrix4f_TanAngleMatrixFromProjection(&projectionMatrix);
+
+  if (externalSurfaceId == DEFAULT_EXTERNAL_SURFACE_ID) {
+    for (int i = 0; i < VRAPI_FRAME_LAYER_EYE_MAX; ++i) {
+      const auto &eyeSwapChain = m.eyeSwapChains[i];
+      const int swapChainIndex = m.frameIndex % eyeSwapChain->swapChainLength;
+      // Set up OVR layer textures
+      projection.Textures[i].ColorSwapChain = eyeSwapChain->ovrSwapChain;
+      projection.Textures[i].SwapChainIndex = swapChainIndex;
+      projection.Textures[i].TexCoordsFromTanAngles = ovrMatrix4f_TanAngleMatrixFromProjection(&projectionMatrix);
+    }
+  } else {
+    const int swapChainIndex = externalSurfaceId;
+    // Flip texCoord in vertical when using WebGL frame textures.
+    const ovrMatrix4f texFlip = ovrMatrix4f_CreateScale(1, -1, 1);
+    ovrMatrix4f proj = ovrMatrix4f_Multiply(&projectionMatrix, &texFlip);
+    proj = ovrMatrix4f_TanAngleMatrixFromProjection(&proj);
+
+    for (int i = 0; i < DeviceDelegateOculusVR::State::EYE_SURFACE_COUNT; ++i) {
```

Comment:
`VRAPI_EYE_COUNT` should be used for the loop instead of the `EYE_SURFACE_COUNT`. We need to set the values for both eyes, if the EYE_SURFACE_COUNT was set to > 2 it could overflow.